systemlayer/executionrequests.go

```diff
@@ -8,32 +8,58 @@ import (
 	pilosa "github.com/featurebasedb/featurebase/v3"
 )
 
+const (
+	keepLastRequests int = 2000
+	truncateTextAt   int = 4096
+)
+
 // ExecutionRequests is an internal struct that keeps a list of sql execution requests
 // this data allows visbility into queries that have been run and are running
 type ExecutionRequests struct {
 	sync.RWMutex
-	requests map[string]*pilosa.ExecutionRequest
+	requestsList []string
+	requests     map[string]*pilosa.ExecutionRequest
 }
 
 // Ensure type implements interface.
 var _ pilosa.ExecutionRequestsAPI = (*ExecutionRequests)(nil)
 
 func NewExecutionRequestsAPI() *ExecutionRequests {
 	return &ExecutionRequests{
-		requests: make(map[string]*pilosa.ExecutionRequest),
+		requestsList: make([]string, 0),
+		requests:     make(map[string]*pilosa.ExecutionRequest),
 	}
 }
 
 // AddRequest adds a new request to the ExecutionRequests struct
-// TODO(pok) ensure a cap on these so we don't suck up too much memory
 func (e *ExecutionRequests) AddRequest(requestID string, userID string, startTime time.Time, sql string) error {
 	e.Lock()
 	defer e.Unlock()
 
+	// we're going to add a new request so clear out oldest request if we've hit
+	// the threshhold
+	if len(e.requestsList) >= keepLastRequests {
+		// get the oldest request
+		delId := e.requestsList[0]
+
+		// make sure it's not this request, unlikely, but
+		// we've all been bitten by 'this should never happen' before
+		// and if it isn't delete it from the map and the list
+		if delId != requestID {
```

Comment:
why do we care if the oldest request has the same ID as this request? Won't we just add this request to the map and list again below?

Reply:
Good point - was being overly cautious...

Reply:
I think the current code might actually be worse as if all the request IDs were the same somehow the list grows indefinitely again.